L/libcxxwrap_julia/build_tarballs.jl

```diff
@@ -2,39 +2,49 @@
 # `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
 
+julia_version = v"1.3.1"
+
 name = "libcxxwrap_julia"
-version = v"0.8.0"
+version = VersionNumber("0.8.2-julia.$(julia_version.major).$(julia_version.minor)")
```

Comment:
What's the rationale of this version number?  Pkg doesn't really like anything outside of X.Y.Z and we overstress it with the build number.

Reply:
Here's the reason: We need to build separate binaries (artifacts) of libcxxwrap for *at least Julia 1.3 versus Julia >= 1.4; but better, for each of Julia 1.3, 1.4, 1.5, 1.6.  Each of the corresponding JLL wrappers needs a different `compat` entry for Julia in `Project.toml`. But to do that, they need actual different versions (the build number part after the `+` is not enough for that, the registry doesn't support that, which makes sense, as it is consistent with how semver defines build number).

Unlike build numbers, the pre-release is part of semver which is explicitly defined to indicate different versions, including how they should be compared with each other and other versions. It also is used by Julia itself As such, I would hope that existing Julia tooling correctly deals with it (if not, that'd be a bug in the semver support of that tool, which IMHO should be fixed). 

One drawback is that the semver spec says this "A pre-release version indicates that the version is unstable and might not satisfy  the intended compatibility requirements as denoted by its associated normal version." However, that should not matter for any tooling, it's just something about the semantics for users; and the only user of this JLL would be CxxWrap.jl, so there should be no issue here, I think.

Reply:
I think this is OK, as long as it works. Not knowing about pre-releases, what I had in mind for this was:
* libcxxwrap 0.8.2 = Julia 1.3
* libcxxwrap 0.8.3 = Julia 1.4
* libcxxwrap 0.8.4 = Julia 1.5
* libcxxwrap 0.8.5 = Julia 1.6

Starting from Julia 1.6, I am hoping this is solved by the julia version being part of the artifact resolution, if I understood correctly?

Reply:
> Starting from Julia 1.6, I am hoping this is solved by the julia version being part of the artifact resolution, if I understood correctly?

Well, that would work for Julia v1.6+

Reply:
@barche if you are willing to go through that hassle, sure, why not (but perhaps shift all the libcxxwrap version up by one, then the last digits of the libcxxwrap and Julia version match, that might make things easier?).

Of course it also raises the question how you then would version a bugfix release? Perhaps by adding 10 to the patchlevel, so that Julia 0.8.13 = Julia 1.3,  etc., then 0.8.23 = Julia 1.3 etc?

Reply:
@dilumaluthge just clarified on Slack: "We don't allow prereleases in the General registry."

Reply:
@barche so, what's the plan now? Do you want to take over this PR, or make a new one? (Feel free to use any or all of what's in here, or also feel free to directly modify this PR as you like)

Reply:
Sorry for the slow response. I propose to leave the version number at ` v"0.8.0"`, I don't really plan any bugfix releases since this is a temporary situation until Julia 1.6 is out, would that be acceptable?

Reply:
? A single version wouldn't allow to produce several binaries for several Julia versions, though? And hence wouldn't allow to resolve the crashes with Julia 1.3.

Also, the current release is 0.8.0, hence this should be at least 0.8.1.

It is also far from clear to me that this is a temporary situation which will go away with Julia 1.6 (plus, it will likely be months until then, won't it? : first off, that assumes everyone is ready to immediately drop support for all Julia versions once 1.6.0 is released. Secondly, it is not yet clear to me that Julia 1.6 will really resolve this: are all pieces already there? Even if, without verifying that it works as desired in practice, I'd be reluctant to rely on that (no offense intended to @staticfloat and anybody else  who may habe worked on this).

Reply:
No, what I mean is keep the proposed mapping as:

* libcxxwrap 0.8.2 = Julia 1.3
* libcxxwrap 0.8.3 = Julia 1.4
* libcxxwrap 0.8.4 = Julia 1.5
* libcxxwrap 0.8.5 = Julia 1.6

If you think bugfixes are important for older Julia versions, we could do:

* libcxxwrap 0.8.2 = Julia 1.3
* libcxxwrap 0.9.0 = Julia 1.4
* libcxxwrap 0.10.0 = Julia 1.5
* libcxxwrap 0.11.0 = Julia 1.6

Reply:
As long as they are all using the same source, I'd suggest to keep them all at 0.8.x

Reply:
I've changed the version (back) to 0.8.2, as discussed here. Can this be merged now, or is anything still to be done? (I am not aware of anything, but...)

Reply:
OK for me!